### ISSUE TITLE
Bugs within rSprite.sdLimits

### ISSUE THREAD

**Issue by LukasWallrich** [sTeamTraen/rSPRITE#1]
When working on [rsprite2](lukaswallrich/rsprite2), we found two issues with the rSprite.sdLimits, both in the following block

```r
  if ((diff < 0) && (k > 1)) {
      vec <- c(rep(a, k - 1), abs(diff), rep(b, N - k))
    }
    else if ((diff > 0) && ((N - k) > 1)) {
      vec <- c(rep(a, k), diff, rep(b, N - k - 1))
    }
```

Firstly, just adding diff and abs(diff) into the vector often leads to vecs that have the wrong mean, and thus SD- for instance, if you start with N = 5, tMean = 4.2, scaleMin = 1, scaleMax = 7, then the maximum SD is calculated based on c(1, 1, 2, 7, 7) because diff = -2. Instead, it should be calculated based on c(1, 1, 7-2, 7, 7). So I think a and b need to feature in the adjusted term as below:

```r
  if ((diff < 0) && (k > 1)) {
      vec <- c(rep(a, k - 1), a + abs(diff), rep(b, N - k))
    }
    else if ((diff > 0) && ((N - k) > 1)) {
      vec <- c(rep(a, k), b - diff, rep(b, N - k - 1))
    }
```

Furthermore, the terms requiring k or N-k to be greater than 1 *force* a and b to remain in the vector. This can lead to errors a) when the lower limit can be 0 and b) when the distribution cannot contain either extreme value - for instance when the underlying distribution is c(20,  9,  4, 35,  6, 16) on a scale from 3 to 84,  i.e. N = 6, tMean = 15, scaleMin = 3, scaleMax = 84, the distribution cannot actually contain 84, so that the returned upper limit is wrong. Given that k is >=1, I think these conditionals can just go, resulting in

```r
  if ((diff < 0) ) {
      vec <- c(rep(a, k - 1), a + abs(diff), rep(b, N - k))
    }
    else if ((diff > 0) ) {
      vec <- c(rep(a, k), b - diff, rep(b, N - k - 1))
    }
```

Hope that helps.

**Comment by sTeamTraen** [comment]
Hi Lukas,

This is excellent, thanks. I will run some checks and add this to the
production code of rSPRITE if it doesn't break anything. Then I will try to
remember how Git works to push the changes!

Regarding the second point, I suspect that I included the minimum length of
1 for each rep() out of an unfounded fear that if the length was zero, the
resulting non-vector would give problems with concatenated with the other
two. But a quick manual check shows that this isn't a problem.

Best,
Nick

**Comment by sTeamTraen** [comment]
Fixed in v0.18.
